src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,121 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IReadOnlyList<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+            return TranslatePaths(storageFiles);
         }
 
         public override IEnumerable<FileSystemInfo> EnumerateFileSystemInfos(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new FileSystemInfo[0];
+            IReadOnlyList<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+            return TranslateFileSystemInfos(storageFiles);
+        }
+
+        /// <summary>
+        /// Translates IStorageItems into FileSystemInfos
+        /// </summary>
+        private static IEnumerable<FileSystemInfo> TranslateFileSystemInfos(IReadOnlyList<IStorageItem> storageFiles)
```

Comment:
Nit: I find the `Translate` prefix a little confusing at first glance as I'm not sure what it's *translating*. How about using the same naming pattern Linq uses for its iterator methods: public methods like `Select` do the up-front argument validation and then call a second private method named `SelectIterator` (suffixed with `Iterator`) which does the lazy yielding. So `EnumerateFileSystemInfosIterator` instead of `TranslateFileSystemInfos`?